src/shared/utils/wait.ts

```diff
@@ -1,23 +1,28 @@
-import { GolemTimeoutError } from "../error/golem-error";
+import { GolemAbortError, GolemTimeoutError } from "../error/golem-error";
 
 /**
  * Utility function that helps to block the execution until a condition is met (check returns true) or the timeout happens.
  *
  * @param {function} check - The function checking if the condition is met.
  * @param {Object} [opts] - Options controlling the timeout and check interval in seconds.
- * @param {number} [opts.timeoutSeconds=30] - The timeout value in seconds.
+ * @param {number} [opts.signalOrTimeout=30] - The timeout value in seconds or AbortSignal.
  * @param {number} [opts.intervalSeconds=1] - The interval between condition checks in seconds.
  *
  * @return {Promise<void>} - Resolves when the condition is met or rejects with a timeout error if it wasn't met on time.
  */
 export function waitForCondition(
   check: () => boolean | Promise<boolean>,
-  opts?: { timeoutSeconds?: number; intervalSeconds?: number },
+  opts?: { signalOrTimeout?: number | AbortSignal; intervalSeconds?: number },
 ): Promise<void> {
-  const timeoutSeconds = opts?.timeoutSeconds ?? 30;
+  const DEFAULT_TIMEOUT_SECONDS = 30;
+  const abortSignal =
```

Comment:
can't we use `createAbortSignalFromTimeout` here instead of these nested ternaries?
```ts
  const signal =
    opts?.signalOrTimeout !== undefined
      ? createAbortSignalFromTimeout(opts?.signalOrTimeout)
      : AbortSignal.timeout(DEFAULT_TIMEOUT_SECONDS * 1000);
```

Reply:
I don't think so, because `opts?.signalOrTimeout` is given in sec instead of ms and we still have to add checking if there is a number and multiply by 1000, so I don't know what is better...?

Reply:
`signalOrTimeout` should be in milliseconds then, let's be consistent across the codebase with this

Reply:
ok, we agree 🙂